### PR TITLE
Extend backoff retries to Segment auth calls

### PR DESCRIPTION
**Pull request by bmedx**

### CODE REVIEW

File: tubular/segment_api.py [hunk 1]
```diff
@@ -144,6 +144,7 @@ def _get_auth_token(self):
         resp_json = None
         try:
             resp = requests.post(self.base_url, json=mutation)
+            resp.raise_for_status()
```
Comment (doctoryes) [inline tubular/segment_api.py:147]: I don't understand why this method call was added?

File: tubular/segment_api.py [hunk 2]
```diff
@@ -144,6 +144,7 @@ def _get_auth_token(self):
...
             resp_json = resp.json()
             return resp_json['data']['login']['access_token']
         except (TypeError, KeyError, JSONDecodeError):
```
Comment (pwnage101) [inline tubular/segment_api.py]: Should this now also catch and log HTTPError?

File: tubular/segment_api.py (hunk 1 above)
Comment (doctoryes) [reply tubular/segment_api.py:147]: Oh - nm - I see that this is a `requests` thing.

File: tubular/segment_api.py (hunk 2 above)
Comment (doctoryes) [reply tubular/segment_api.py]: Agreed - from `requests` doc:
Response.raise_for_status() will raise an HTTPError if the HTTP request returned an unsuccessful status code.

File: tubular/tests/test_segment_api.py [hunk 3]
```diff
@@ -114,26 +134,24 @@ def test_bulk_delete_error(setup_bulk_delete, caplog):  # pylint: disable=redefi
...
+                'fake_base_url', 'fake_email', 'fake_password', 'fake_workspace'
+            ]]
+        )
 
-                learner = TEST_SEGMENT_CONFIG['learner']
+        learner = TEST_SEGMENT_CONFIG['learner']
 
-                with pytest.raises(JSONDecodeError):
-                    segment.delete_learners(learner, 1000)
+        with pytest.raises(JSONDecodeError):
```
Comment (pwnage101) [inline tubular/tests/test_segment_api.py]: How come this works for FakeAuthErrorResponse?  Isn't that one supposed to raise HTTPError due to the `resp.raise_for_status()`?

Comment (bmedx) [reply tubular/tests/test_segment_api.py]: Updated to make the fakery more realistic

**Review by pwnage101 — APPROVED** [review]
Approved, but you can't fool me---definitely fake.

**Pull request merged**